### PR TITLE
Add anchor underline to high contrast

### PR DESCRIPTION
**Pull request by dancormier**
Upstream changes to support https://github.com/StackEng/StackOverflow/pull/9535

Only merge once https://github.com/StackEng/StackOverflow/pull/9535 and https://github.com/StackEng/StackOverflow/pull/9526 are merged.

### CODE REVIEW

File: lib/css/base/_stacks-reset-normalize.less [hunk 1]
```diff
@@ -99,11 +99,14 @@ pre {
...
  * 1. Remove the gray background on active links in IE 10.
  * 2. Remove gaps in links underline in iOS 8+ and Safari 8+.
+ * 3. Adds underline to links in high contrast
  */
 
 a {
   background-color: transparent; /* 1 */
   -webkit-text-decoration-skip: objects; /* 2 */
+
+  .highcontrast-mode({ text-decoration: underline; });
```
Comment (aaronshekey) [inline lib/css/base/_stacks-reset-normalize.less]: Let's do this in `stacks-links.less` instead: https://github.com/StackExchange/Stacks/blob/develop/lib/css/components/_stacks-links.less#L17

There's a future where we kill these resets entirely #609

Comment (dancormier) [reply lib/css/base/_stacks-reset-normalize.less]: Good call. I knew it was wrong when I was doing it and I did it anyway. 🤷‍♂️

File: lib/css/base/_stacks-reset-normalize.less [hunk 2]
```diff
@@ -99,6 +99,7 @@ pre {
 /**
  * 1. Remove the gray background on active links in IE 10.
  * 2. Remove gaps in links underline in iOS 8+ and Safari 8+.
+ * 3. Adds underline to links in high contrast
```
Comment (aaronshekey) [inline lib/css/base/_stacks-reset-normalize.less]:
```suggestion
```

**Review by aaronshekey — CHANGES_REQUESTED** [review]
Just 2 teeny tiny bits. Merge whenever you'd like into production SO. That work can definitely ship before the next release of Stacks.

We've got a blue that's a little leaky in the case of `s-card`. 
image

**Review by aaronshekey — APPROVED** [review]

**Pull request merged**